src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionViewModel.cs

```diff
@@ -183,15 +187,56 @@ internal sealed class ConnectionViewModel : BindableBase, IConnectionViewModel {
             IsRemote = !(uri?.IsFile ?? true);
         }
 
-       public string GetCompletePath() {
+        public void UpdateName() {
+            var currentPath = Path?.Trim() ?? string.Empty;
+            var previousPath = _previousPath?.Trim() ?? string.Empty;
+
+            var previousProposedName = GetProposedName(previousPath);
+            var currentProposedName = GetProposedName(currentPath);
+
+            // Avoid changing anything if the edit
+            // has no effect on the name (like changing the path's port)
+            if (previousProposedName != currentProposedName) {
+                // Check if the name was calculated from the previous path
+                var currentName = Name ?? string.Empty;
+                if (string.IsNullOrEmpty(currentName) || string.Compare(currentName, previousProposedName, StringComparison.CurrentCultureIgnoreCase) == 0) {
```

Comment:
But also, does this need to ignore case? It feels like the only time when it'd be different is if the user manually edited it to make the case differ, in which case it might be better to not get in the way.